docs/usage.rst

```diff
@@ -0,0 +1,162 @@
+.. Specify pandas format output in cells
+.. jupyter-execute::
+    :hide-code:
+    :hide-output:
+
+    import pandas as pd
+
+
+    def series_to_html(self):
+        df = self.to_frame()
+        df.columns = ['']
+        return df._repr_html_()
+
+
+    setattr(pd.Series, '_repr_html_', series_to_html)
+    pd.set_option('display.max_rows', 6)
+
+.. Specify version for storing and loading objects to YAML
+.. jupyter-execute::
+    :hide-code:
+
+    __version__ = '1.0.0'
+
+
+Usage
+=====
+
+The basic idea of :mod:`audinterface` is
+to provide easy and standardized interfaces
+to apply a machine learning model,
+or other digital signal processing algorithms (DSP)
+to audio files.
+The only prerequisite is
+the algorithm provides a callable
+that takes at least the signal
+as a :class:`np.ndarray`
+and the sampling rate as input.
+
+The interface can then apply the algorithm
+on a list of files,
+a folder,
+or an index conform to the audformat_ database specification.
+Results are always returned containing a `segmented index`_.
+In the following we load three files from the emodb_ database
+and define a list of files,
+a folder,
+and an index.
+
+.. jupyter-execute::
+
+    import audb
+    import os
+
+    media = ['wav/03a01Fa.wav', 'wav/03a01Nc.wav', 'wav/03a01Wa.wav']
+    db = audb.load('emodb', version='1.2.0', media=media, verbose=False)
+
+    files = list(db.files)
+    folder = os.path.dirname(files[0])
+    index = db['emotion'].index
+
+
+DSP algorithm returning a series
+--------------------------------
+
+Let's assume we want to calculate the root mean square (RMS)
+value in dB.
+We first define the function
+and create an interface for it using :class:`audinterface.Process`.
+
+.. jupyter-execute::
+
+    import audinterface
+    import numpy as np
+
+    def rms(signal, sampling_rate):
+        return 20 * np.log10(np.sqrt(np.mean(signal ** 2)))
+
+    interface = audinterface.Process(process_func=rms)
+
+The following three commands
+apply the algorithm
+and all return the same result
+as a :class:`pd.Series`.
+
+.. jupyter-execute::
+
+    y = interface.process_files(files)
+    y = interface.process_folder(folder)
+    y = interface.process_index(index)
+    y
+
+
+DSP algorithm returning a dataframe
+-----------------------------------
+
+When using a DSP algorithm as feature extractor,
+it is recommended to use :class:`audinterface.Feature`,
+which returns results as a :class:`pd.DataFrame`
+and assigns names to the single features.
```

Comment:
Another advantage is that we can set a window and hop duration in case of features with dynamic length. I think we should highlight by adding a second example.

Reply:
I added such an example under the new subsection "Create a framewise feature extractor".
But maybe my implementation of calculating the mean on a framewise basis is slightly clumsy ;)

![image](https://user-images.githubusercontent.com/173624/172867156-942bc33a-426f-4d4f-a4cf-236fefb5c698.png)

Reply:
Maybe:

```python
def features(signal, sampling_rate, win_dur, hop_dur):
    win_dur = int(win_dur * sampling_rate)
    hop_dur = int(hop_dur * sampling_rate)
    starts = np.arange(0, signal.shape[1] - win_dur, hop_dur)
    mean = [[signal[:, s:s+win_dur].mean() for s in starts]]
    return mean
```

Reply:
Or we switch to an example where we interface an external library, e.g.:

```python
win_dur = 0.02
hop_dur = 0.01
n_mfcc = 13
interface = audinterface.Feature(
    [f'mfcc-{idx}' for idx in range(n_mfcc)],
    process_func=features,
    process_func_args={
        'win_dur': win_dur,
        'hop_dur': hop_dur,
        'n_mfcc': n_mfcc,
    },
    hop_dur=hop_dur,
    win_dur=win_dur,
)
df = interface.process_files(db.files[:10])
df
```
```
                                                                                                      mfcc-0  ...    mfcc-12
file                                               start                  end                                 ...           
/media/jwagner/Data/audb/emodb/1.2.0/fe182b91/w... 0 days 00:00:00        0 days 00:00:00.020000 -611.993286  ...   1.151396
                                                   0 days 00:00:00.010000 0 days 00:00:00.030000 -668.175842  ...  14.068544
                                                   0 days 00:00:00.020000 0 days 00:00:00.040000 -664.612793  ...   7.949759
                                                   0 days 00:00:00.030000 0 days 00:00:00.050000 -667.714722  ...  12.957479
                                                   0 days 00:00:00.040000 0 days 00:00:00.060000 -669.367432  ...   4.396849
...                                                                                                      ...  ...        ...
/media/jwagner/Data/audb/emodb/1.2.0/fe182b91/w... 0 days 00:00:01.650000 0 days 00:00:01.670000 -619.806335  ...  11.462805
                                                   0 days 00:00:01.660000 0 days 00:00:01.680000 -622.713928  ...  13.530191
                                                   0 days 00:00:01.670000 0 days 00:00:01.690000 -624.977356  ...  12.032556
                                                   0 days 00:00:01.680000 0 days 00:00:01.700000 -621.033264  ...  11.019306
                                                   0 days 00:00:01.690000 0 days 00:00:01.710000 -622.040344  ...  11.202772

[1743 rows x 13 columns]
```

Reply:
I think it's general a good idea to cover this kind of use-case.

Reply:
Good idea, I switched to `librosa.feature.mfcc`